[OSDOCS 7033]: Update Network Verification sections (#63341)
openshift/openshift-docs: modules/running-network-verification-manually-cli.adoc

.Procedure

* Verify the network configuration by using one of the following methods:
** Verify the network configuration by specifying the cluster name. The subnet IDs are automatically detected:
+
[source,terminal]
----
$ rosa verify network --cluster <cluster_name> <1>
----
<1> Replace `<cluster_name>` with the name of your cluster.
+
.Example output
[source,terminal]
----
I: Verifying the following subnet IDs are configured correctly: [subnet-0b761d44d3d9a4663 subnet-0f87f640e56934cbc]
I: subnet-0b761d44d3d9a4663: pending
I: subnet-0f87f640e56934cbc: passed
----
*** Ensure that verification to all subnets has been completed.
+
[source,terminal]
----
$ rosa verify network --watch <1>
                      --status-only <2>
                      --region us-east-1 <3>

>                       --subnet-ids subnet-0b761d44d3d9a4663,subnet-0f87f640e56934cbc <4>

Reviewer: I copy the command but it is not valid.
```
rosa verify network --watch 
Error: unknown flag: --watch
Usage:
  rosa verify [command]

Available Commands:
  openshift-client Verify OpenShift client tools
  permissions      Verify AWS permissions are ok for non-STS cluster install
  quota            Verify AWS quota is ok for cluster install

Flags:
  -h, --help   help for verify

Global Flags:
      --color string   Surround certain characters with escape sequences to display them in color on the terminal. Allowed options are [auto never always] (default "auto")
      --debug          Enable debug mode.

Use "rosa verify [command] --help" for more information about a command.

Failed to execute root command: unknown flag: --watch
[yingzhan@localhost rosa]$                       --status-only 
bash: --status-only: command not found...
[yingzhan@localhost rosa]$                       --region us-east-1 
bash: --region: command not found...
[yingzhan@localhost rosa]$                       --subnet-ids subnet-0b761d44d3d9a4663,subnet-0f87f640e56934cbc 
bash: --subnet-ids: command not found...

```

Author: Did you copy the entire command? 
`$ rosa verify network --watch --status-only --region us-east-1 --subnet-ids subnet-0b761d44d3d9a4663,subnet-0f87f640e56934cbc`.

This is an example output from the SME:

```[user]$ rosa verify network --cluster xxxx
I: Verifying the following subnet IDs are configured correctly: [subnet-0b761d44d3d9a4663 subnet-0f87f640e56934cbc]
I: subnet-0b761d44d3d9a4663: pending
I: subnet-0f87f640e56934cbc: passed
I: Run the following command to wait for verification to all subnets to complete:
rosa verify network --watch --status-only --region us-east-1 --subnet-ids subnet-0b761d44d3d9a4663,subnet-0f87f640e56934cbc
[user]$ rosa verify network --watch --status-only --region us-east-1 --subnet-ids subnet-0b761d44d3d9a4663,subnet-0f87f640e56934cbc
I: Checking the status of the following subnet IDs: [subnet-0b761d44d3d9a4663 subnet-0f87f640e56934cbc]
I: subnet-0f87f640e56934cbc: passed
I: subnet-0b761d44d3d9a4663: passed
[user]$ rosa verify network --watch --status-only --region us-east-1 --subnet-ids subnet-0b761d44d3d9a4663,subnet-0f87f640e56934cbc
I: Checking the status of the following subnet IDs: [subnet-0b761d44d3d9a4663 subnet-0f87f640e56934cbc]
I: subnet-0b761d44d3d9a4663: passed
I: subnet-0f87f640e56934cbc: passed
[thomasmckay@thomasmckay rosa{master}]$ rosa verify network  --status-only --region us-east-1 --subnet-ids subnet-0b761d44d3d9a4663,subnet-0f87f640e56934cbc
I: Checking the status of the following subnet IDs: [subnet-0b761d44d3d9a4663 subnet-0f87f640e56934cbc]
I: subnet-0b761d44d3d9a4663: passed
I: subnet-0f87f640e56934cbc: passed
[user]$ rosa verify network  --status-only --cluster tomckay
I: Checking the status of the following subnet IDs: [subnet-0b761d44d3d9a4663 subnet-0f87f640e56934cbc]
I: subnet-0b761d44d3d9a4663: passed
I: subnet-0f87f640e56934cbc: passed

```

Reviewer: I click the back Copy icon and then paste it to run.  I paste the content here
```
rosa verify network --watch 
                      --status-only 
                      --region us-east-1 
                      --subnet-ids subnet-0b761d44d3d9a4663,subnet-0f87f640e56934cbc 
```

Author: Hi @thomasmckay. Can you please advise on this? Thanks.

Reviewer 2: I'm not sure I understand the question. The docs don't have line continuation characters (`\`) but instead have numbers for documentation (`<1>`).

Author: Thanks @thomasmckay. I added the `\` as they should have been there but @yingzhanredhat says the command does not work.

Author: Hi @yingzhanredhat, I ran the following commands and they worked for me: 
```
$ rosa verify network --cluster aecollin
I: Verifying the following subnet IDs are configured correctly: [subnet-02b865f670e8283ef subnet-0d33a8516675d1a4c]
I: subnet-02b865f670e8283ef: pending
I: subnet-0d33a8516675d1a4c: passed
I: Run the following command to wait for verification to all subnets to complete:
rosa verify network --watch --status-only --subnet-ids subnet-02b865f670e8283ef,subnet-0d33a8516675d1a4c
```  
```
$ rosa verify network --watch --status-only --region us-east-1 --subnet-ids subnet-02b865f670e8283ef,subnet-0d33a8516675d1a4c
I: Checking the status of the following subnet IDs: [subnet-02b865f670e8283ef subnet-0d33a8516675d1a4c]
I: subnet-0d33a8516675d1a4c: passed
I: subnet-02b865f670e8283ef: passed
```

Author: @yingzhanredhat Can you please review the amended command? Many thanks.

----

> <1> The `watch` flag causes the command to not run until all the subnets under test are in either a failed or passed state.

Reviewer 2: `...not complete...` (I'm not sure what a good word here is. The `watch` arg is common to other commands so maybe there's another example.)

Author: Thanks. Have updated.

> <1> The `watch` flag causes the command to not run until all the subnets under test are in either a failed or passed state.

Reviewer 3: ```suggestion
<1> The `watch` flag causes the command to run after all the subnets under test are in a failed or passed state.
```
A suggestion for minimalism.

Author: Thanks Tami - have implemented all feedback!

> <2> The `status-only` flag does not trigger a run of network verification but returns the current state that is pending, for example, `subnet-123 (verification still in-progress)`. By default, without this option, a call to this command always triggers a verification of the specified subnets.

Reviewer: This description is not correct. It will return the current status. Not necessarily "pending"
> but returns the current state that is pending

Author: Thanks I will change this.

> <2> The `status-only` flag does not trigger a run of network verification but returns the current state which is pending, for example, `subnet-123 (verification still in-progress)`. By default, without this option, a call to this command always triggers a verification of the specified subnets.

Reviewer 3: ```suggestion
<2> The `status-only` flag does not trigger a run of network verification but returns the current state that is pending, for example, `subnet-123 (verification still in-progress)`. By default, without this option, a call to this command always triggers a verification of the specified subnets.
```
See ISG: https://www.ibm.com/docs/en/ibm-style?topic=grammar-pronouns#relative-pronouns

<3> Use a specific AWS region, which overrides the _AWS_REGION_ environment variable.

> <4> Enter a list of subnet IDs separated by commas to verify. If any of the subnets do not exist, the command exits with error code one (1) and no subnets are checked.

Reviewer: What does the "(1)" means？

Author: It is the error code. I did not format it correctly. I will change that.

Author: Updated error message.

+
.Example output
[source,terminal]
----
I: Checking the status of the following subnet IDs: [subnet-0b761d44d3d9a4663 subnet-0f87f640e56934cbc]
I: subnet-0f87f640e56934cbc: passed
I: subnet-0b761d44d3d9a4663: passed

> I: ✓ Network verification successful

Reviewer: I don't see this info "I: ✓ Network verification successful" after I run the command

```
./rosa verify network  --status-only --region us-west-2 --subnet-ids subnet-082410d8fb3d95195,subnet-0fb407a4dc8974e87,subnet-0700cf0d732aef010,subnet-0febb5c2620e3548e,subnet-0ba62f961cac934f9,subnet-0f8ac35d20f1d659d
I: Checking the status of the following subnet IDs: [subnet-082410d8fb3d95195 subnet-0fb407a4dc8974e87 subnet-0700cf0d732aef010 subnet-0febb5c2620e3548e subnet-0ba62f961cac934f9 subnet-0f8ac35d20f1d659d]
I: subnet-082410d8fb3d95195: passed
I: subnet-0fb407a4dc8974e87: passed
I: subnet-0700cf0d732aef010: passed
I: subnet-0febb5c2620e3548e: passed
I: subnet-0ba62f961cac934f9: passed
I: subnet-0f8ac35d20f1d659d: passed
```

Author: Thanks, I will update this.

> I don't see this info "I: ✓ Network verification successful" after I run the command
> 
> ```
> ./rosa verify network  --status-only --region us-west-2 --subnet-ids subnet-082410d8fb3d95195,subnet-0fb407a4dc8974e87,subnet-0700cf0d732aef010,subnet-0febb5c2620e3548e,subnet-0ba62f961cac934f9,subnet-0f8ac35d20f1d659d
> I: Checking the status of the following subnet IDs: [subnet-082410d8fb3d95195 subnet-0fb407a4dc8974e87 subnet-0700cf0d732aef010 subnet-0febb5c2620e3548e subnet-0ba62f961cac934f9 subnet-0f8ac35d20f1d659d]
> I: subnet-082410d8fb3d95195: passed
> I: subnet-0fb407a4dc8974e87: passed
> I: subnet-0700cf0d732aef010: passed
> I: subnet-0febb5c2620e3548e: passed
> I: subnet-0ba62f961cac934f9: passed
> I: subnet-0f8ac35d20f1d659d: passed
> ```

Thanks I will change this.

<3> Use a specific AWS region that overrides the _AWS_REGION_ environment variable.
<4> Enter a list of subnet IDs separated by commas to verify. If any of the subnets do not exist, the error message `Network verification for subnet 'subnet-<subnet_number> not found` displays and no subnets are checked.
+
.Example output
[source,terminal]
----
I: Checking the status of the following subnet IDs: [subnet-03146b9b52b6024cb subnet-03146b9b52b2034cc]
I: subnet-03146b9b52b6024cb: passed
I: subnet-03146b9b52b2034cc: passed
----
+
[TIP]
====
To output the full list of verification tests, you can include the `--debug` argument when you run the `rosa verify network` command.
====
+
** Verify the network configuration by specifying the VPC subnets IDs, replacing `<region_name>` with your AWS region:
+
[source,terminal]
----
$ rosa verify network --status-only --region <region_name> --subnet-ids subnet-03146b9b52b6024cb,subnet-03146b9b52b2034cc
----
+
.Example output
[source,terminal]
----

> $ rosa verify network --region us-east-1 --status-only --subnet-ids subnet-03146b9b52b6024cb,subnet-03146b9b52b2034cc

Reviewer: The command is not correct.
1. The command should not contain "status-only" as the flag does not trigger a run of network verification
2. Check  network verification via subnets command should like this : rosa verify network \
      --subnet-ids subnet-0b761d44d3d9a4663,subnet-0f87f640e56934cbc  \
      --role-arn arn:aws:iam::765374464689:role/my-Installer-Role

rosa verify network --region us-east-1 --status-only --subnet-ids subnet-03146b9b52b6024cb,subnet-03146b9b52b2034cc

Author: > The command is not correct.
> 
> 1. The command should not contain "status-only" as the flag does not trigger a run of network verification
> 2. Check  network verification via subnets command should like this : rosa verify network 
>    --subnet-ids subnet-0b761d44d3d9a4663,subnet-0f87f640e56934cbc  
>    --role-arn arn:aws:iam::765374464689:role/my-Installer-Role
> 
> rosa verify network --region us-east-1 --status-only --subnet-ids subnet-03146b9b52b6024cb,subnet-03146b9b52b2034cc

Thanks @yingzhanredhat . I have amended this command.

Author: @yingzhanredhat When I try to run this command, I get an error message. It seems the region is required: 
```
$ rosa verify network --subnet-ids subnet-07e040767f6a7dcaa,subnet-03090605cc9803257 --role-arn arn:aws:iam::269733383066:role/ManagedOpenShift-Installer-Role
E: Region is required

``` 
The command is successful with the `--region` flag: 
```
$ rosa verify network --subnet-ids subnet-07e040767f6a7dcaa,subnet-03090605cc9803257 --region us-east-1 --role-arn arn:aws:iam::269733383066:role/ManagedOpenShift-Installer-Role
I: Verifying the following subnet IDs are configured correctly: [subnet-07e040767f6a7dcaa subnet-03090605cc9803257]
I: subnet-07e040767f6a7dcaa: passed
I: subnet-03090605cc9803257: pending
I: Run the following command to wait for verification to all subnets to complete:
rosa verify network --watch --status-only --subnet-ids subnet-07e040767f6a7dcaa,subnet-03090605cc9803257

```

Author: Hi @thomasmckay. Can you please advise on whether `--region` should be used as part of this command? Thanks.

Reviewer 2: yes, `--region` is required

Author: Thanks @thomasmckay.
